### PR TITLE
Add parameter constraints to IAU disk-integrated phase function models

### PR DESCRIPTION
**Pull request by jianyangli**

### CODE REVIEW

File: docs/sbpy/photometry.rst [hunk 1]
```diff
@@ -96,37 +96,43 @@ model parameters:
...
+initialized with the first set of valid parameters in the input
+`~sbpy.data.DataClass`.
 
 To fit a photometric model, one may follow the standard procedure defined in
 `astropy.modeling` submodule, first initializing a model, then using one of
 the fitter classes defined in `astropy.modeling.fitting`
-submodule, such as `~astropy.modeling.fitting.LevMarLSQFitter`.
+submodule, such as `~astropy.modeling.fitting.SLSQPLSQFitter`.  Note that
+the `~sbpy.photometry.HG1G2` model requires a constraint, 1 - G1 - G2 < 1,
+that is implemented in sbpy as an ineqcons-type constraint (see
```
Comment (mkelley) [inline docs/sbpy/photometry.rst:108]: Could you write this more plainly, and move "ineqcons" to the remark?  e.g.:

"Note that  the `~sbpy.photometry.HG1G2` model requires an inequality constraint, 1 - G1 - G2 < 1, implemented in sbpy via the `ineqcons` parameter of `~astropy.modeling.Model`."

File: CHANGES.rst [hunk 2]
```diff
@@ -12,10 +12,18 @@ sbpy.data
...
 - Added ``DataClass.__contains__`` to enable `in` operator for ``DataClass``
   objects. [#357]
 
 
+sbpy.photometry
+^^^^^^^^^^^^^^^
+
+- Added parameter constraints to the IAU disk-integrated phase function models,
+  such as ``HG``, ``HG1G2``, ``HG12``, and ``HG12_Pen16``. [#366]
+
```
Comment (mkelley) [inline CHANGES.rst:25]: Also, note change API: the NonmonotonicPhaseFunctionWarning has been replaced with InvalidPhaseFunctionWarning

**Review by mkelley — COMMENTED** [review]
Sorry for the delay.  Just a couple comments.

File: docs/sbpy/photometry.rst (hunk 1 above)
Comment (jianyangli) [reply docs/sbpy/photometry.rst]: updated.

**Comment by jianyangli** [comment]
@mkelley This PR is now ready for review again.  See if it looks good to you.  Thanks.

File: docs/sbpy/photometry.rst [hunk 3]
```diff
@@ -96,37 +96,45 @@ model parameters:
...
 `astropy.modeling` submodule, first initializing a model, then using one of
 the fitter classes defined in `astropy.modeling.fitting`
-submodule, such as `~astropy.modeling.fitting.LevMarLSQFitter`.
+submodule, such as `~astropy.modeling.fitting.SLSQPLSQFitter`.  Note that
+the `~sbpy.photometry.HG1G2` model requires that the G1 and G2 parameters
+are bounded within [0, 1], as well as an inequality constraint,
+0 <= 1 - G1 - G2 <= 1.  These constraints are implemented in sbpy via the
+`bounds` parameter of `~astropy.modeling.Parameter` and the `ineqcons`
+parameter of `~astropy.modeling.Model`.  Some fitters, such as
+`astropy.modeling.LevMarLSQFitter`, doesn't support constrained fit via
```
[the remark this answers is not in the document]
Comment (jianyangli) [reply docs/sbpy/photometry.rst]: Thanks.  Fixed now.

**Pull request merged**